woocommerce/woocommerce
# Product Gallery block: Add logic to trap keyboard focus within the Product Gallery Pop-Up

### Changes proposed in this Pull Request:




When opening the Product Gallery pop-up, keyboard focus is not trapped within the pop-up itself. This allows users to accidentally navigate away from the pop-up using keyboard commands, potentially leading to confusion and accessibility challenges.

This PR adds the logic to trap the keyboard focus within the Product Gallery pop-up.

Closes #43762 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Single Product template (Appearance > Editor > Templates > Single Product template).
2. Click on "Edit" to open the Template editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block." Click on it.
4. In the block inserter, search for "Product Gallery" or browse through the available blocks until you find it.
5. Click on the "Product Gallery" block to add it to your content.
6. Visit a product with multiple images.
7. Click on the large image to open the Product Gallery Pop-Up;
8. Make sure the focus is on the Previous Image button;
9. Press 'Tab' multiple times and ensure the focus is always within the Product Gallery Pop-Up.
10. Press 'Shift + Tab' multiple times to navigate backward and ensure the focus is always within the Product Gallery Pop-Up.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Add focus trapping within the Product Gallery Pop-Up to enhance keyboard navigation and accessibility.

#### Comment 

</details>
